src/libraries/Hooks.sol

```diff
@@ -83,6 +91,147 @@ library Hooks {
+        IHooks self,
+        PoolKey memory key,
+        IPoolManager.ModifyPositionParams memory params,
+        bytes calldata hookData
+    ) internal returns (bool noop) {
+        if (self.shouldCallBeforeModifyPosition()) {
+            noop = self.callHookNoopable(
+                abi.encodeWithSelector(IHooks.beforeModifyPosition.selector, msg.sender, key, params, hookData)
+            );
+        }
+    }
+
+    /// @notice calls afterModifyPosition hook if permissioned and validates return value
+    function afterModifyPosition(
+        IHooks self,
+        PoolKey memory key,
+        IPoolManager.ModifyPositionParams memory params,
+        BalanceDelta delta,
+        bytes calldata hookData
+    ) internal {
+        if (key.hooks.shouldCallAfterModifyPosition()) {
+            self.callHook(
+                abi.encodeWithSelector(IHooks.afterModifyPosition.selector, msg.sender, key, params, delta, hookData)
+            );
+        }
+    }
+
+    /// @notice calls beforeSwap hook if permissioned and validates return value
+    function beforeSwap(IHooks self, PoolKey memory key, IPoolManager.SwapParams memory params, bytes calldata hookData)
+        internal
+        returns (bool noop)
+    {
+        if (key.hooks.shouldCallBeforeSwap()) {
+            noop = self.callHookNoopable(
+                abi.encodeWithSelector(IHooks.beforeSwap.selector, msg.sender, key, params, hookData)
+            );
+        }
+    }
+
+    /// @notice calls afterSwap hook if permissioned and validates return value
+    function afterSwap(
+        IHooks self,
+        PoolKey memory key,
+        IPoolManager.SwapParams memory params,
+        BalanceDelta delta,
+        bytes calldata hookData
+    ) internal {
+        if (key.hooks.shouldCallAfterSwap()) {
+            self.callHook(abi.encodeWithSelector(IHooks.afterSwap.selector, msg.sender, key, params, delta, hookData));
+        }
+    }
+
+    /// @notice calls beforeDonate hook if permissioned and validates return value
+    function beforeDonate(IHooks self, PoolKey memory key, uint256 amount0, uint256 amount1, bytes calldata hookData)
+        internal
+        returns (bool noop)
+    {
+        if (key.hooks.shouldCallBeforeDonate()) {
+            noop = self.callHookNoopable(
+                abi.encodeWithSelector(IHooks.beforeDonate.selector, msg.sender, key, amount0, amount1, hookData)
+            );
+        }
+    }
+
+    /// @notice calls afterDonate hook if permissioned and validates return value
+    function afterDonate(IHooks self, PoolKey memory key, uint256 amount0, uint256 amount1, bytes calldata hookData)
+        internal
+    {
+        if (key.hooks.shouldCallAfterDonate()) {
+            self.callHook(
+                abi.encodeWithSelector(IHooks.afterDonate.selector, msg.sender, key, amount0, amount1, hookData)
+            );
+        }
+    }
+
     function shouldCallBeforeInitialize(IHooks self) internal pure returns (bool) {
```

Comment:
maybe should remove these `shouldCall*` things and just inline them in the hook functions themselves

Reply:
agreed done